src/components/calendar/calendar.jsx

```diff
@@ -0,0 +1,199 @@
+import React, {
+  useState,
+} from 'react';
+import PropTypes from 'prop-types';
+import arrowLeft from '../../svgs/arrow-left.svg';
+import arrowRight from '../../svgs/arrow-right.svg';
+import caretDown from '../../svgs/caret-down.svg';
+import IconButton from '../icon-button/icon-button.jsx';
+import Icon from '../icon/icon.jsx';
+import styles from './calendar.css';
+
+function getDateIterator(year, month) {
+  const firstDate = new Date(year, month, 1);
+  return new Date(year, month, 1 - firstDate.getDay());
+}
+
+function getHeadCell(iterator) {
+  const date = iterator.getDate();
+  const weekday = iterator.toLocaleDateString(undefined, {
+    weekday: 'narrow',
+  });
+
+  // This needs to occur at the end of this function
+  iterator.setDate(date + 1);
+
+  return (
+    <th className={styles.weekday}>
+      {weekday}
+    </th>
+  );
+}
+
+function getCellClassName(iterator, month, highlightedDate) {
+  if (
+    highlightedDate.getFullYear() === iterator.getFullYear()
+    && highlightedDate.getMonth() === iterator.getMonth()
+    && highlightedDate.getDate() === iterator.getDate()
+  ) return styles['highlighted-date'];
+
+  if (iterator.getMonth() === month) return styles.date;
+
+  return styles['not-current-date'];
+}
+
+function getCell(iterator, month, highlightedDate) {
+  const date = iterator.getDate();
+  const className = getCellClassName(iterator, month, highlightedDate);
+
+  // This needs to occur at the end of this function
```

Comment:
I'm pretty cool with this comment and the one above, but would it be possible to include the behavior this comment points out in a test? If so, would it be safe to remove this comment/allow `iterator.setDate` to occur in a different order of other future changes? Either way, a spec ensuring this behavior would be ideal, although a comment like this is significant enough for me to let it go.

Reply:
Yeah, a test would be ideal. Lemme see if I can pull something together for it.

Reply:
I just realized we already have test coverage with our snapshots tests. I'll go ahead and delete these!